src/azul/indexer/__init__.py

```diff
@@ -77,25 +83,62 @@ def __str__(self) -> str:
 
 
 @attr.s(frozen=True, auto_attribs=True, kw_only=True)
-class SimpleSourceName(SourceName['SimpleSourceName']):
+class SimpleSourceSpec(SourceSpec['SimpleSourceSpec']):
     """
     Default implementation for unstructured source names.
     """
     name: str
 
     @classmethod
-    def parse(cls, name: str) -> 'SimpleSourceName':
-        return cls(name=name)
+    def parse(cls, spec: str) -> 'SimpleSourceSpec':
+        """
+        >>> SimpleSourceSpec.parse('https://foo.edu:12')
+        SimpleSourceSpec(prefix='12', name='https://foo.edu')
+
+        >>> SimpleSourceSpec.parse('foo:bar/baz:12')
+        SimpleSourceSpec(prefix='12', name='foo:bar/baz')
+
+        >>> SimpleSourceSpec.parse('foo')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Source specs must contain a prefix delimiter
+
+        >>> SimpleSourceSpec.parse('foo:8F53')
+        Traceback (most recent call last):
+        ...
+        azul.uuids.InvalidUUIDPrefixError: '8F53' is not a valid UUID prefix.
+
+        >>> SimpleSourceSpec.parse('https://foo.edu')
+        Traceback (most recent call last):
+        ...
+        azul.uuids.InvalidUUIDPrefixError: '//foo.edu' is not a valid UUID prefix.
+        """
+
+        # FIXME: Move parsing of prefix to SourceSpec
+        #        https://github.com/DataBiosphere/azul/issues/3073
+        name, sep, prefix = spec.rpartition(':')
+        require(bool(sep), 'Source specs must contain a prefix delimiter')
```

Comment:
```suggestion
        reject(sep is None, 'Source specifications must end in a colon followed by an optional UUID prefix}')
```

potentially wrapped.

Reply:
https://docs.python.org/3/library/stdtypes.html#str.rpartition
```
 If the separator is not found, return a 3-tuple containing two empty strings, followed by the string itself.
```

Going to use condition `sep is ''`